### PR TITLE
Bug 1559869 - prefer secrets named worker-pool:..

### PR DESCRIPTION
**Pull request by djmitche**

### CODE REVIEW

File: secrets/secrets_test.go [hunk 1]
```diff
@@ -63,6 +63,19 @@ func TestGetSecretConfigFilesFormat(t *testing.T) {
 	assert.Equal(t, true, run.WorkerConfig.MustGet("from-secret"), "value for from-secret")
 }
 
+func TestGetSecretConfigFilesFormat(t *testing.T) {
+	runnercfg, run := setup(t)
+
+	tc.FakeSecretsCreateSecret("worker-pool:pp/wt", &tcsecrets.Secret{
+		Secret: []byte(`{"config": {"from-secret": true}}`),
```
Comment (milescrabill) [inline secrets/secrets_test.go]: Should we put `files` in here or wait?

Comment (djmitche) [reply secrets/secrets_test.go]: That's a stlil-to-be-implemented feature, so "wait".

File: README.md [hunk 2]
```diff
@@ -94,7 +94,10 @@ parameters for a run of a worker:
    step provides Taskcluster credentials for the worker, as well as
    identification information (worker pool, worker ID, etc.).
  * Using the Taskcluster credentials, load configuration from the secrets
-   service with secret name `worker-type:<workerPoolId>`.
+   service with secret name `worker-pool:<workerPoolId>` (and, for
+   compatibility, supports secrets named `worker-type:<workerPoolId>` too, and
+   secrets containing the expected `{config: .., files: ..}` or for
+   compatibility just the worker configuration).
```
[the remark this answers is not in the document]
Comment (djmitche) [reply README.md]: I'll explain it better in the comments.  There's two backward-compatibility things here.

File: secrets/secrets_test.go (hunk 1 above)
Comment (djmitche) [reply secrets/secrets_test.go]: https://bugzilla.mozilla.org/show_bug.cgi?id=1561957

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (milescrabill) [reply README.md]: Yeah, after reading the code it made sense, should have updated.

**Pull request merged**